### ISSUE TITLE
CHOLMOD solver error

### ISSUE THREAD

**Issue by amatem** [Circuitscape/Circuitscape.jl#300]
Julia-related error: Not sure if this is specific to the CHOLMOD solver, the version of Circuitscape or Julia in general but I received the following error after running ~ 5000 pair solves using pairwise mode. I previously haven't had issues generating results with this many solves in the past, so I'm curious what went wrong here. Thanks! (image below)

---

It appears this may be an issue with a package dependency that provides the CHOLMOD solver

INPUTS: Pairwise, 115 nodes
![Capture2](https://user-images.githubusercontent.com/82612877/121221579-8e8a0380-c83a-11eb-993c-8da3e579bf7f.PNG)

OUTPUT: 
![Capture (1)](https://user-images.githubusercontent.com/82612877/121221703-ab263b80-c83a-11eb-8746-8b02e9a621f1.PNG)

**Comment by ranjanan** [comment]
Could you please post some files here (through a drive link) to reproduce this?

**Comment by amatem** [comment]
> Could you please post some files here (through a drive link) to reproduce this?

https://drive.google.com/file/d/1lhFoizjrqp19vK6SRKWYtyKBKY87WdNY/view?usp=sharing
Thank you for helping, here are the ini and raster files I used. Hope this is what you'll need.

**Comment by amatem** [comment]
small update: this solver works ok with a simple 1 pair solve output I tested, but several times with trying greater than 5000 pair solves the Julia environment closes out all together with no output after several hours of run time

![image](https://user-images.githubusercontent.com/82612877/121710295-524ee100-ca8e-11eb-8a0e-d382ce449775.png)

**Comment by vlandau** [comment]
@amatem if it's stopping without any error message and Julia is closing entirely, that could be due to memory constraints on the computer you're using. In that case, you might try using the CG+AMG solver instead of CHOLMOD, as CG+AMG uses less memory (but takes a bit more time)

**Comment by vlandau** [comment]
@amatem any updates?

**Comment by amatem** [comment]
@vlandau Thanks for checking in, project was put on hold and will rerun later this month or next with both solvers and also will try on a higher-RAM computer we just acquired

**Comment by amatem** [comment]
After a long hiatus, I am back to running Circuitscape tests.

I've had success running some of the previous files several months ago using cholmod and this issue was closed out. However the issue returned this week when I switched to a higher RAM research computer, ran parallelization, and used the cholmod solver for pairwise mode on 6,786 pair solves and 117 nodes.
This run occurred on the same sized areasand with the same number of nodes I've previously had successful outputs with however this time I used a an even larger matrix (15 meters/pixel resolution instead of 30 meters/pixel as previously). I've been receiving this error at the first pair solve again with a note to submit a bug report. If needed I can provide the files I've used and will try again with a different solver to see if the error persists:

julia> compute("I:\\Research\\Shares\\ocamp\\MCT\\individualnuthatch15m\\indivnuthatch117nodes15mt.ini")
[ Info: 2022-02-27 11:37:48 : Precision used: Double
[ Info: 2022-02-27 11:37:48 : Starting up Circuitscape to use 10 processes in parallel
[ Info: 2022-02-27 11:38:18 : Reading maps
[ Info: 2022-02-27 11:40:46 : Resistance/Conductance map has 32909363 nodes
[ Info: 2022-02-27 11:42:38 : Solver used: CHOLMOD
┌ Warning: 2022-02-27 11:42:38 : CHOLMOD only works with double precision. Converting single precision matrix to double
└ @ Circuitscape C:\Users\amtemple\.julia\packages\Circuitscape\Qr6wW\src\logging.jl:25
[ Info: 2022-02-27 11:42:42 : Graph has 32909363 nodes, 117 focal points and 6 connected components
[ Info: 2022-02-27 11:42:45 : Total number of pair solves = 6786
[ Info: 2022-02-27 13:32:26 : Time taken to construct cholesky factor = 6568.8458158
[ Info: 2022-02-27 13:32:43 : Time taken to construct local nodemap = 17.1196915 seconds
[ Info: 2022-02-27 13:35:25 : Solving points 1 to 100
ERROR: ArgumentError: dense matrix construction failed for unknown reasons. Please submit a bug report.
Stacktrace:
  [1] SuiteSparse.CHOLMOD.Dense{Float64}(ptr::Ptr{SuiteSparse.CHOLMOD.C_Dense{Float64}})
    @ SuiteSparse.CHOLMOD C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.6\SuiteSparse\src\cholmod.jl:222
  [2] Dense
    @ C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.6\SuiteSparse\src\cholmod.jl:238 [inlined]
  [3] allocate_dense
    @ C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.6\SuiteSparse\src\cholmod.jl:414 [inlined]
  [4] SuiteSparse.CHOLMOD.Dense{Float64}(A::Matrix{Float64})
    @ SuiteSparse.CHOLMOD C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.6\SuiteSparse\src\cholmod.jl:810
  [5] \
    @ C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.6\SuiteSparse\src\cholmod.jl:1712 [inlined]
  [6] solve_linear_system(factor::SuiteSparse.CHOLMOD.Factor{Float64}, matrix::SparseArrays.SparseMatrixCSC{Float64, Int64}, rhs::Matrix{Float64})
    @ Circuitscape C:\Users\amtemple\.julia\packages\Circuitscape\Qr6wW\src\core.jl:633
  [7] solve(prob::Circuitscape.GraphProblem{Float64, Int64, Circuitscape.CholmodSolver}, solver::Circuitscape.CholmodSolver, flags::Circuitscape.RasterFlags, cfg::Dict{String, String}, log::Bool)
    @ Circuitscape C:\Users\amtemple\.julia\packages\Circuitscape\Qr6wW\src\core.jl:445
  [8] single_ground_all_pairs (repeats 2 times)
    @ C:\Users\amtemple\.julia\packages\Circuitscape\Qr6wW\src\core.jl:65 [inlined]
  [9] _pt_file_no_polygons_path(rasterdata::Circuitscape.RasterData{Float64, Int64}, flags::Circuitscape.RasterFlags, cfg::Dict{String, String})
    @ Circuitscape C:\Users\amtemple\.julia\packages\Circuitscape\Qr6wW\src\raster\pairwise.jl:63
 [10] raster_pairwise(T::Type, V::Type, cfg::Dict{String, String})
    @ Circuitscape C:\Users\amtemple\.julia\packages\Circuitscape\Qr6wW\src\raster\pairwise.jl:29
 [11] _compute(T::Type, V::Type, cfg::Dict{String, String})
    @ Circuitscape C:\Users\amtemple\.julia\packages\Circuitscape\Qr6wW\src\run.jl:43
 [12] macro expansion
    @ .\timing.jl:287 [inlined]
 [13] compute(path::String)
    @ Circuitscape C:\Users\amtemple\.julia\packages\Circuitscape\Qr6wW\src\run.jl:31
 [14] top-level scope
    @ REPL[5]:1

**Comment by ViralBShah** [comment]
When this happens, can you keep a watch on the memory utilization of your computer? If it is almost full, it means you are trying to solve a much larger problem than you have memory for.

The other thing is you can run this without parallelism, which will use lesser memory and see how far it goes.

**Comment by amatem** [comment]
Thank you! You are absolutely right, I checked it three times and looks like memory hits >90% right when this error message occurs. I switched to the cg+amg and it seems to be running well so far. Silly question, but are there any significant differences in the output cumulative current maps using different solvers (cholmod vs cg+amg?) that could occur or other major differences between choosing one method over the other in particular circumstances.

**Comment by amatem** [comment]
Here's the full print out

[ Info: 2022-02-28 12:34:56 : Scheduling pair 116 of 6786 to be solved
[ Info: 2022-02-28 12:34:56 : Solving pair 1 of 6786
[ Info: 2022-02-28 12:40:08 : Time taken to solve linear system = 311.8859266 seconds
ERROR: BoundsError: attempt to access 11-element Vector{SharedArrays.SharedMatrix{Float64}} at index [0]
Stacktrace:
  [1] getindex
    @ .\array.jl:805 [inlined]
  [2] maybeview
    @ .\views.jl:134 [inlined]
  [3] dotview
    @ .\broadcast.jl:1212 [inlined]
  [4] write_cur_maps(name::String, output::Circuitscape.Output{Float64, Int64}, component_data::Circuitscape.ComponentData{Float64, Int64}, finitegrounds::Vector{Float64}, flags::Circuitscape.RasterFlags, cfg::Dict{String, String})
    @ Circuitscape C:\Users\amtemple\.julia\packages\Circuitscape\QbA2E\src\out.jl:98
  [5] macro expansion
    @ .\timing.jl:287 [inlined]
  [6] postprocess(output::Circuitscape.Output{Float64, Int64}, component_data::Circuitscape.ComponentData{Float64, Int64}, flags::Circuitscape.RasterFlags, shortcut::Circuitscape.Shortcut{Float64}, cfg::Dict{String, String})
    @ Circuitscape C:\Users\amtemple\.julia\packages\Circuitscape\QbA2E\src\core.jl:667
  [7] (::Circuitscape.var"#f#100"{Float64, Int64, Circuitscape.RasterFlags, Dict{String, String}, Bool, Circuitscape.Shortcut{Float64}, Circuitscape.Cumulative{Float64, Int64}, Vector{Int64}, Vector{Tuple{Int64, Int64}}, Vector{Int64}, Vector{Int64}, Circuitscape.ComponentData{Float64, Int64}, AlgebraicMultigrid.Preconditioner{AlgebraicMultigrid.MultiLevel{AlgebraicMultigrid.Pinv{Float64}, AlgebraicMultigrid.GaussSeidel{AlgebraicMultigrid.SymmetricSweep}, AlgebraicMultigrid.GaussSeidel{AlgebraicMultigrid.SymmetricSweep}, SparseArrays.SparseMatrixCSC{Float64, Int64}, SparseArrays.SparseMatrixCSC{Float64, Int64}, LinearAlgebra.Adjoint{Float64, SparseArrays.SparseMatrixCSC{Float64, Int64}}, AlgebraicMultigrid.MultiLevelWorkspace{Vector{Float64}, 1}}, AlgebraicMultigrid.V}, SparseArrays.SparseMatrixCSC{Float64, Int64}, Vector{Int64}})(i::Int64)
    @ Circuitscape C:\Users\amtemple\.julia\packages\Circuitscape\QbA2E\src\core.jl:231
  [8] #98
    @ C:\Users\amtemple\.julia\packages\Circuitscape\QbA2E\src\core.jl:251 [inlined]
  [9] iterate
    @ .\generator.jl:47 [inlined]
 [10] _collect(c::UnitRange{Int64}, itr::Base.Generator{UnitRange{Int64}, Circuitscape.var"#98#104"{Circuitscape.var"#f#100"{Float64, Int64, Circuitscape.RasterFlags, Dict{String, String}, Bool, Circuitscape.Shortcut{Float64}, Circuitscape.Cumulative{Float64, Int64}, Vector{Int64}, Vector{Tuple{Int64, Int64}}, Vector{Int64}, Vector{Int64}, Circuitscape.ComponentData{Float64, Int64}, AlgebraicMultigrid.Preconditioner{AlgebraicMultigrid.MultiLevel{AlgebraicMultigrid.Pinv{Float64}, AlgebraicMultigrid.GaussSeidel{AlgebraicMultigrid.SymmetricSweep}, AlgebraicMultigrid.GaussSeidel{AlgebraicMultigrid.SymmetricSweep}, SparseArrays.SparseMatrixCSC{Float64, Int64}, SparseArrays.SparseMatrixCSC{Float64, Int64}, LinearAlgebra.Adjoint{Float64, SparseArrays.SparseMatrixCSC{Float64, Int64}}, AlgebraicMultigrid.MultiLevelWorkspace{Vector{Float64}, 1}}, AlgebraicMultigrid.V}, SparseArrays.SparseMatrixCSC{Float64, Int64}, Vector{Int64}}}}, #unused#::Base.EltypeUnknown, isz::Base.HasShape{1})
    @ Base .\array.jl:695
 [11] collect_similar(cont::UnitRange{Int64}, itr::Base.Generator{UnitRange{Int64}, Circuitscape.var"#98#104"{Circuitscape.var"#f#100"{Float64, Int64, Circuitscape.RasterFlags, Dict{String, String}, Bool, Circuitscape.Shortcut{Float64}, Circuitscape.Cumulative{Float64, Int64}, Vector{Int64}, Vector{Tuple{Int64, Int64}}, Vector{Int64}, Vector{Int64}, Circuitscape.ComponentData{Float64, Int64}, AlgebraicMultigrid.Preconditioner{AlgebraicMultigrid.MultiLevel{AlgebraicMultigrid.Pinv{Float64}, AlgebraicMultigrid.GaussSeidel{AlgebraicMultigrid.SymmetricSweep}, AlgebraicMultigrid.GaussSeidel{AlgebraicMultigrid.SymmetricSweep}, SparseArrays.SparseMatrixCSC{Float64, Int64}, SparseArrays.SparseMatrixCSC{Float64, Int64}, LinearAlgebra.Adjoint{Float64, SparseArrays.SparseMatrixCSC{Float64, Int64}}, AlgebraicMultigrid.MultiLevelWorkspace{Vector{Float64}, 1}}, AlgebraicMultigrid.V}, SparseArrays.SparseMatrixCSC{Float64, Int64}, Vector{Int64}}}})
    @ Base .\array.jl:606
 [12] map(f::Function, A::UnitRange{Int64})
    @ Base .\abstractarray.jl:2294
 [13] solve(prob::Circuitscape.GraphProblem{Float64, Int64, Circuitscape.AMGSolver}, #unused#::Circuitscape.AMGSolver, flags::Circuitscape.RasterFlags, cfg::Dict{String, String}, log::Bool)
    @ Circuitscape C:\Users\amtemple\.julia\packages\Circuitscape\QbA2E\src\core.jl:251
 [14] single_ground_all_pairs (repeats 2 times)
    @ C:\Users\amtemple\.julia\packages\Circuitscape\QbA2E\src\core.jl:66 [inlined]
 [15] _pt_file_no_polygons_path(rasterdata::Circuitscape.RasterData{Float64, Int64}, flags::Circuitscape.RasterFlags, cfg::Dict{String, String})
    @ Circuitscape C:\Users\amtemple\.julia\packages\Circuitscape\QbA2E\src\raster\pairwise.jl:63
 [16] raster_pairwise(T::Type, V::Type, cfg::Dict{String, String})
    @ Circuitscape C:\Users\amtemple\.julia\packages\Circuitscape\QbA2E\src\raster\pairwise.jl:29
 [17] _compute(T::Type, V::Type, cfg::Dict{String, String})
    @ Circuitscape C:\Users\amtemple\.julia\packages\Circuitscape\QbA2E\src\run.jl:43
 [18] macro expansion
    @ .\timing.jl:287 [inlined]
 [19] compute(path::String)
    @ Circuitscape C:\Users\amtemple\.julia\packages\Circuitscape\QbA2E\src\run.jl:31
 [20] top-level scope
    @ REPL[11]:1

**Comment by vlandau** [comment]
> are there any significant differences in the output cumulative current maps using different solvers (cholmod vs cg+amg?) that could occur or other major differences between choosing one method over the other in particular circumstances.

The solver is basically the algorithm that is used to solve a complex linear algebra problem. Often, it's too large to be solved directly, which is where an iterative approach like cg+amg is used (as opposed to cholmod, which solves the problem directly but requires a lot of memory to do so). Both cg+amg and cholmod are just different methods to solve the same problem, so any difference should be extremely minimal. I guess that was a long-winded way to say no, there shouldn't be any difference 🙂

**Comment by amatem** [comment]
I spoke too soon, looks like I hit a new error after switching to a different solver (see edited post). Thanks so much for all your help explaining the solvers.

**Comment by vlandau** [comment]
@amatem Can you post the full stack trace? (all of the output when the program errored)

**Comment by amatem** [comment]
Is this what you need?

julia>  compute("I:\\Research\\Shares\\ocamp\\MCT\\individualnuthatch15m\\indivnuthatch117nodes15mt1.ini")
[ Info: 2022-02-28 12:50:32 : Precision used: Double
[ Info: 2022-02-28 12:50:32 : Reading maps
[ Info: 2022-02-28 12:52:51 : Resistance/Conductance map has 32909363 nodes
[ Info: 2022-02-28 12:54:35 : Solver used: AMG accelerated by CG
[ Info: 2022-02-28 12:54:35 : Graph has 32909363 nodes, 117 focal points and 6 connected components
[ Info: 2022-02-28 12:54:37 : Total number of pair solves = 6786
[ Info: 2022-02-28 12:55:55 : Time taken to construct preconditioner = 65.364584 seconds
[ Info: 2022-02-28 12:56:09 : Time taken to construct local nodemap = 13.4412149 seconds
[ Info: 2022-02-28 12:56:09 : Scheduling pair 1 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 2 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 3 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 4 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 5 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 6 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 7 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 8 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 9 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 10 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 11 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 12 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 13 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 14 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 15 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 16 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 17 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 18 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 19 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 20 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 21 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 22 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 23 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 24 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 25 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 26 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 27 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 28 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 29 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 30 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 31 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 32 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 33 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 34 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 35 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 36 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 37 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 38 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 39 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 40 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 41 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 42 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 43 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 44 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 45 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 46 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 47 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 48 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 49 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 50 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 51 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 52 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 53 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 54 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 55 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 56 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 57 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 58 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 59 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 60 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 61 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 62 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 63 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 64 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 65 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 66 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 67 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 68 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 69 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 70 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 71 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 72 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 73 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 74 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 75 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 76 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 77 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 78 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 79 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 80 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 81 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 82 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 83 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 84 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 85 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 86 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 87 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 88 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 89 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 90 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 91 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 92 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 93 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 94 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 95 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 96 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 97 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 98 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 99 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 100 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 101 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 102 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 103 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 104 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 105 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 106 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 107 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 108 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 109 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 110 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 111 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 112 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 113 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 114 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 115 of 6786 to be solved
[ Info: 2022-02-28 12:56:09 : Scheduling pair 116 of 6786 to be solved
[ Info: 2022-02-28 12:56:10 : Solving pair 1 of 6786
[ Info: 2022-02-28 13:01:18 : Time taken to solve linear system = 307.9571439 seconds
ERROR: BoundsError: attempt to access 11-element Vector{SharedArrays.SharedMatrix{Float64}} at index [0]
Stacktrace:
  [1] getindex
    @ .\array.jl:805 [inlined]
  [2] maybeview
    @ .\views.jl:134 [inlined]
  [3] dotview
    @ .\broadcast.jl:1212 [inlined]
  [4] write_cur_maps(name::String, output::Circuitscape.Output{Float64, Int64}, component_data::Circuitscape.ComponentData{Float64, Int64}, finitegrounds::Vector{Float64}, flags::Circuitscape.RasterFlags, cfg::Dict{String, String})
    @ Circuitscape C:\Users\amtemple\.julia\packages\Circuitscape\QbA2E\src\out.jl:98
  [5] macro expansion
    @ .\timing.jl:287 [inlined]
  [6] postprocess(output::Circuitscape.Output{Float64, Int64}, component_data::Circuitscape.ComponentData{Float64, Int64}, flags::Circuitscape.RasterFlags, shortcut::Circuitscape.Shortcut{Float64}, cfg::Dict{String, String})
    @ Circuitscape C:\Users\amtemple\.julia\packages\Circuitscape\QbA2E\src\core.jl:667
  [7] (::Circuitscape.var"#f#100"{Float64, Int64, Circuitscape.RasterFlags, Dict{String, String}, Bool, Circuitscape.Shortcut{Float64}, Circuitscape.Cumulative{Float64, Int64}, Vector{Int64}, Vector{Tuple{Int64, Int64}}, Vector{Int64}, Vector{Int64}, Circuitscape.ComponentData{Float64, Int64}, AlgebraicMultigrid.Preconditioner{AlgebraicMultigrid.MultiLevel{AlgebraicMultigrid.Pinv{Float64}, AlgebraicMultigrid.GaussSeidel{AlgebraicMultigrid.SymmetricSweep}, AlgebraicMultigrid.GaussSeidel{AlgebraicMultigrid.SymmetricSweep}, SparseArrays.SparseMatrixCSC{Float64, Int64}, SparseArrays.SparseMatrixCSC{Float64, Int64}, LinearAlgebra.Adjoint{Float64, SparseArrays.SparseMatrixCSC{Float64, Int64}}, AlgebraicMultigrid.MultiLevelWorkspace{Vector{Float64}, 1}}, AlgebraicMultigrid.V}, SparseArrays.SparseMatrixCSC{Float64, Int64}, Vector{Int64}})(i::Int64)
    @ Circuitscape C:\Users\amtemple\.julia\packages\Circuitscape\QbA2E\src\core.jl:231
  [8] #98
    @ C:\Users\amtemple\.julia\packages\Circuitscape\QbA2E\src\core.jl:251 [inlined]
  [9] iterate
    @ .\generator.jl:47 [inlined]
 [10] _collect(c::UnitRange{Int64}, itr::Base.Generator{UnitRange{Int64}, Circuitscape.var"#98#104"{Circuitscape.var"#f#100"{Float64, Int64, Circuitscape.RasterFlags, Dict{String, String}, Bool, Circuitscape.Shortcut{Float64}, Circuitscape.Cumulative{Float64, Int64}, Vector{Int64}, Vector{Tuple{Int64, Int64}}, Vector{Int64}, Vector{Int64}, Circuitscape.ComponentData{Float64, Int64}, AlgebraicMultigrid.Preconditioner{AlgebraicMultigrid.MultiLevel{AlgebraicMultigrid.Pinv{Float64}, AlgebraicMultigrid.GaussSeidel{AlgebraicMultigrid.SymmetricSweep}, AlgebraicMultigrid.GaussSeidel{AlgebraicMultigrid.SymmetricSweep}, SparseArrays.SparseMatrixCSC{Float64, Int64}, SparseArrays.SparseMatrixCSC{Float64, Int64}, LinearAlgebra.Adjoint{Float64, SparseArrays.SparseMatrixCSC{Float64, Int64}}, AlgebraicMultigrid.MultiLevelWorkspace{Vector{Float64}, 1}}, AlgebraicMultigrid.V}, SparseArrays.SparseMatrixCSC{Float64, Int64}, Vector{Int64}}}}, #unused#::Base.EltypeUnknown, isz::Base.HasShape{1})
    @ Base .\array.jl:695
 [11] collect_similar(cont::UnitRange{Int64}, itr::Base.Generator{UnitRange{Int64}, Circuitscape.var"#98#104"{Circuitscape.var"#f#100"{Float64, Int64, Circuitscape.RasterFlags, Dict{String, String}, Bool, Circuitscape.Shortcut{Float64}, Circuitscape.Cumulative{Float64, Int64}, Vector{Int64}, Vector{Tuple{Int64, Int64}}, Vector{Int64}, Vector{Int64}, Circuitscape.ComponentData{Float64, Int64}, AlgebraicMultigrid.Preconditioner{AlgebraicMultigrid.MultiLevel{AlgebraicMultigrid.Pinv{Float64}, AlgebraicMultigrid.GaussSeidel{AlgebraicMultigrid.SymmetricSweep}, AlgebraicMultigrid.GaussSeidel{AlgebraicMultigrid.SymmetricSweep}, SparseArrays.SparseMatrixCSC{Float64, Int64}, SparseArrays.SparseMatrixCSC{Float64, Int64}, LinearAlgebra.Adjoint{Float64, SparseArrays.SparseMatrixCSC{Float64, Int64}}, AlgebraicMultigrid.MultiLevelWorkspace{Vector{Float64}, 1}}, AlgebraicMultigrid.V}, SparseArrays.SparseMatrixCSC{Float64, Int64}, Vector{Int64}}}})
    @ Base .\array.jl:606
 [12] map(f::Function, A::UnitRange{Int64})
    @ Base .\abstractarray.jl:2294
 [13] solve(prob::Circuitscape.GraphProblem{Float64, Int64, Circuitscape.AMGSolver}, #unused#::Circuitscape.AMGSolver, flags::Circuitscape.RasterFlags, cfg::Dict{String, String}, log::Bool)
    @ Circuitscape C:\Users\amtemple\.julia\packages\Circuitscape\QbA2E\src\core.jl:251
 [14] single_ground_all_pairs (repeats 2 times)
    @ C:\Users\amtemple\.julia\packages\Circuitscape\QbA2E\src\core.jl:66 [inlined]
 [15] _pt_file_no_polygons_path(rasterdata::Circuitscape.RasterData{Float64, Int64}, flags::Circuitscape.RasterFlags, cfg::Dict{String, String})
    @ Circuitscape C:\Users\amtemple\.julia\packages\Circuitscape\QbA2E\src\raster\pairwise.jl:63
 [16] raster_pairwise(T::Type, V::Type, cfg::Dict{String, String})
    @ Circuitscape C:\Users\amtemple\.julia\packages\Circuitscape\QbA2E\src\raster\pairwise.jl:29
 [17] _compute(T::Type, V::Type, cfg::Dict{String, String})
    @ Circuitscape C:\Users\amtemple\.julia\packages\Circuitscape\QbA2E\src\run.jl:43
 [18] macro expansion
    @ .\timing.jl:287 [inlined]
 [19] compute(path::String)
    @ Circuitscape C:\Users\amtemple\.julia\packages\Circuitscape\QbA2E\src\run.jl:31
 [20] top-level scope
    @ REPL[54]:1

**Comment by vlandau** [comment]
Yep, exactly. @ranjanan any idea what's going on?
